gobblin-runtime/src/main/java/gobblin/runtime/TaskState.java

```diff
@@ -101,6 +102,7 @@ public TaskState(WorkUnitState workUnitState) {
     addAll(workUnitState);
     this.jobId = workUnitState.getProp(ConfigurationKeys.JOB_ID_KEY);
     this.taskId = workUnitState.getProp(ConfigurationKeys.TASK_ID_KEY);
+    this.taskKey = workUnitState.getProp(ConfigurationKeys.TASK_KEY_KEY);
```

Comment:
Can you add a default like "unknown_task_key" in this getProp() in case some code path missed setting it? Also, why is the task_key saved? It looks like in all non-test cases the task key is parsed from the task_id, so we can parse it here instead.